rcsb/utils/chemref/DrugBankReader.py

```diff
@@ -173,6 +181,35 @@ def __getTargetInfo(self, category, target):
             )
         ]
         if len(hgncIds) == 1:
-            doc["hgnc_id"] = hgncIds[0][len("HGNC:") :]
+            doc["hgnc_id"] = hgncIds[0][len("HGNC:"):]
 
         return doc
+
+    def __processDrugProducts(self, productL):
+        """Reprocess drug products data to adhere to schema."""
+        dpL = []
+        for pD in productL:
+            dpD = {}
+            for k, v in pD.items():
+                if k == "started-marketing-on":
+                    if v != "":
+                        dpD["started_marketing_on"] = datetime.strptime(v, "%Y-%m-%d")
+                elif k == "ended-marketing-on":
+                    if v != "":
+                        dpD["ended_marketing_on"] = datetime.strptime(v, "%Y-%m-%d")
+                elif k == "approved" and v != "":
+                    if v == "true":
+                        dpD["approved"] = "Y"
+                    elif v == "false":
+                        dpD["approved"] = "N"
+                else:
+                    dpD[k] = v
+            dpL.append(dpD)
+        #
+        drugProductL = self.__removeDuplicateDicts(dpL)
+        return drugProductL
+
+    def __removeDuplicateDicts(self, dictList, key=lambda d: tuple(sorted(d.items()))):
```

Comment:
@piehld Similar to `drug_products`, `drug_groups` also have `uniqueItems` set to `true` in the schema. Perhaps good to address that as well.

Reply:
Thanks @brindakv. I don't think there's a chance `drug_groups` would have duplicate items, but I've added an additional bit of code to ensure that as well.